src/Compiler/FSComp.txt

```diff
@@ -1372,7 +1372,7 @@ tcTupleStructMismatch,"One tuple type is a struct tuple, the other is a referenc
 3201,tcModuleAbbrevFirstInMutRec,"In a recursive declaration group, module abbreviations must come after all 'open' declarations and before other declarations"
 3202,tcUnsupportedMutRecDecl,"This declaration is not supported in recursive declaration groups"
 3203,parsInvalidUseOfRec,"Invalid use of 'rec' keyword"
-3204,tcStructUnionMultiCaseDistinctFields,"If a union type has more than one case and is a struct, then all fields within the union type must be given unique names."
+3204,tcStructUnionMultiCaseDistinctFields,"If a union type has more than one case and is a struct, then all fields within the union type must be given unique field names. For example: 'type A = B of b: int | C of c: int' (unique field names 'b' and 'c' assigned)."
```

Comment:
@T-Gro, @edgarfgp, a bit late to the party, but I'd rather formulate it like this:
> If a multicase union type is a struct, then all union cases must have unique names. For example: 'type A = B of b: int | C of c: int'.`

Reasoning: 
1) Brevity and the accent on the fact it's valid only for structs
2) `must be given` - technically not entirely correct, they just should end up unique in the end, as `type A = | B of b: int | C of int` also works. Therefore `must have`

Also, we could technically point people to the docs, it's well explained there: https://learn.microsoft.com/en-us/dotnet/fsharp/language-reference/discriminated-unions#struct-discriminated-unions